### DISCUSSION TITLE
Bumping Lambda Nodejs versions in Serverless config files

### DISCUSSION THREAD

**Discussion by gustavkj** [renovatebot/renovate#16594] [Suggest an Idea]
I'm currently working on bumping a few repos using [`serverless`](https://github.com/serverless/serverless) to Node 16 and thought it would be nice to get Renovate to help me bump the Node version in the `serverless.yml` file as well.

After reading the docs and looking at how the [`nvm` manager](https://github.com/renovatebot/renovate/tree/8bd77f747b3da59cc955b2fc4ba3301bcb680ea2/lib/modules/manager/nvm) works, the following regex manager is what I have come up with.

I wanted to share this here, if anyone else is trying to solve the same problem. And also, to get input if there are any improvements that can be done to the matcher. 

A few things I was thinking about is; 
- if the version number will remain just the major number?
- if there's a way to limit the allowed Node versions to LTS versions (since [Lambda only supports LTS versions](https://docs.aws.amazon.com/lambda/latest/dg/lambda-nodejs.html))?

```json
{
  "regexManagers": [
    {
      "fileMatch": ["(^|/)serverless.ya?ml$"],
      "matchStrings": ["runtime:\\s+nodejs(?<currentValue>.*?).x\\n"],
      "depNameTemplate": "node",
      "datasourceTemplate": "node",
      "versioningTemplate": "node"
    }
  ]
}
```

**Update:** Changed `datasourceTemplate` to `node` as suggested in the comments. And remove `packageNameTemplate` as it referred to the repo name when using `github-tags` as `datasourceTemplate`. And match config files with either a `.yml` or `.yaml` file extension.

**Comment by viceice** [comment]
- `node` versioning only allows lts by default https://docs.renovatebot.com/modules/versioning/#nodejs-versioning
- You should use `node` datasource https://docs.renovatebot.com/modules/datasource/#node-datasource

**Reply by gustavkj** [reply]
Thank you! I've updated the datasource locally and in the code snippet and good to know the `node` versioning only allows lts.

**Reply by viceice** [reply]
`packageNameTemplate` is now obsolete 😉

**Reply by gustavkj** [reply]
Ah, right. I was a bit confused what it referred to, but now I get that it was used with the `github-tags` datasource to find the repo. I've updated it again. Thanks! 🙇
